vhost_user_net/src/lib.rs

```diff
@@ -93,15 +99,14 @@ impl std::convert::From<Error> for std::io::Error {
 }
 
 pub struct VhostUserNetBackend {
-    mem: Option<GuestMemoryMmap>,
-    vring_worker: Option<Arc<VringWorker>>,
+    mem: Option<Arc<GuestMemoryMmap>>,
+    vring_worker: Option<Vec<Arc<VringWorker>>>,
     kill_evt: EventFd,
     taps: Vec<(Tap, usize)>,
-    rxs: Vec<RxVirtio>,
-    txs: Vec<TxVirtio>,
-    rx_tap_listenings: Vec<bool>,
     num_queues: usize,
     queue_size: u16,
+    vrings: Option<Vec<Arc<RwLock<Vring>>>>,
+    worker_thread: Option<Vec<thread::JoinHandle<VhostUserBackendResult<()>>>>,
```

Comment:
Plural. Does it even need to be an `Option<T>` now it's a `Vec` ?

Reply:
worker_thread in vhost_user_net will be created after memory update, before that,  it should be initialized with None.